webapps-schema/src/main/java/io/camunda/webapps/schema/entities/usermanagement/EntityJoinRelation.java

```diff
@@ -7,40 +7,43 @@
  */
 package io.camunda.webapps.schema.entities.usermanagement;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
 public record EntityJoinRelation(String name, Long parent) {
 
+  @JsonIgnore
+  public String getRouting() {
+    return parent != null ? parent.toString() : null;
+  }
+
   public static class EntityJoinRelationFactory {
 
-    private final IdentityJoinRelationshipType parent;
-    private final IdentityJoinRelationshipType child;
+    private final String parentName;
+    private final String childName;
 
-    public EntityJoinRelationFactory(
-        final IdentityJoinRelationshipType parent, final IdentityJoinRelationshipType child) {
-      this.parent = parent;
-      this.child = child;
+    public EntityJoinRelationFactory(final String parentName, final String childName) {
+      this.parentName = parentName;
+      this.childName = childName;
     }
 
     public EntityJoinRelation createParent() {
-      return new EntityJoinRelation(parent.getType(), null);
+      return new EntityJoinRelation(parentName, null);
     }
 
     public EntityJoinRelation createChild(final long parentKey) {
-      return new EntityJoinRelation(child.getType(), parentKey);
+      return new EntityJoinRelation(childName, parentKey);
     }
-  }
-
-  public enum IdentityJoinRelationshipType {
-    GROUP("group"),
-    MEMBER("member");
```

Comment:
There is string repetiton in the [GroupIndex](https://github.com/camunda/camunda/pull/24609/files#diff-61b32049479cc0f4d1f47a77e6cb16821fe18818fc9dc2149719a131b7c342ecR19) and [RoleIndex](https://github.com/camunda/camunda/pull/24609/files#diff-b1529f330014dfe26200f492ceae14fc9c3999d18607958db41fa2d2e6356ca1R23). So the String for the "member" join role will be repeated for the Role, Group, Tenant, and Authorization indexes. 

Since the original goal of this refactoring was to reduce the probability of error, this change introduces more probabilities of typos, and it makes it more difficult to track down the issue if it happens.

Reply:
These two strings are equal, but that's accidental. One refers to the child name of the parent-child relation in the group index, the other refers to the name of the relation in the role index. Those don't have to be equal.

Reply:
Elsewhere in the code we have opted to use ENUMs instead of strings even if their reuse is limited/singular. although we only use these once currently I'd leave them as an ENUM so we are consistent and if we need them for whatever reason in the future its one less thing

Reply:
Ok, if you both agree that we should have the enum I can revert the refactoring.

I believe the code is strictly worse with the enum though as that suggests that it's ok to refer to them directly. The point of the factory abstraction is to make that unnecessary: you can refer to the parent or child name without knowing which is which by calling `getChildName()`/`getParentName()`.